bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/widgets/Shell.java

```diff
@@ -479,20 +479,42 @@ void addToolTip (ToolTip toolTip) {
 	toolTips = newToolTips;
 }
 
-void adjustTrim () {
+/**
+ * Resize the shell based on the trim width and height. Trim width and height
+ * adjustments are read from 'user/.swt.trims.prefs' for each trim style. If
+ * such a file does not exist then it uses default values initialized.
+ *
+ * Note that we are trying to resize the shell which is already open. Shell
+ * already has width and height, Which is either set by setSize()/setBounds() or
+ * size is calculated based on layout and content of the shell. Without setting
+ * bounds/layout shell won't display content which is created on it.
+ *
+ * @param curWidth  Current width of the shell that is already open.
+ * @param curHeight Current height of the shell that is alreday open.
+ */
+void adjustTrim (int curWidth, int curHeight) {
```

Comment:
`curWidth, curHeight` - I believe we should name it `widthHint`and `heightHint`, as those are not necessarily the correct sizes (depends on actual trim)